modules/infra/control/control_output.c

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2024 Christophe Fontaine
+
+#include <gr_control.h>
+#include <gr_control_output.h>
+#include <gr_graph.h>
+#include <gr_log.h>
+#include <gr_macro.h>
+#include <gr_mbuf.h>
+
+#include <event2/event.h>
+#include <rte_ether.h>
+#include <rte_graph_worker.h>
+
+#include <pthread.h>
+#include <stdatomic.h>
+#include <stdlib.h>
+
+extern struct rte_ring *control_output_ring;
+static struct event *control_output_evt;
+
+static int tshutdown;
+pthread_t thread_id;
+pthread_cond_t cv;
+pthread_mutex_t mp;
+
+static void poll_control_output_ring(evutil_socket_t, short, void *) {
+	struct gr_control_output_msg msg[RTE_GRAPH_BURST_SIZE];
+	int n = 0;
+
+	while ((n = rte_ring_dequeue_burst_elem(
+			control_output_ring, msg, sizeof(msg[0]), ARRAY_DIM(msg), NULL
+		))
+	       > 0) {
+		for (int i = 0; i < n; i++) {
+			msg[i].callback(msg[i].mbuf);
+		}
+	}
+}
+
+void signal_control_ouput_message() {
+	pthread_cond_signal(&cv);
+}
+
+static void *cond_wait_to_event(void *) {
+	struct timespec ts;
+	pthread_setname_np(pthread_self(), "gr:cond_wait_to_event");
```

Comment:
`cond_wait_to_event` seems overly focused on the implementation and not on the actual purpose of that thread. How about `"gr:ctrl-output"`?